closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -162,12 +173,22 @@ protected Key engineDoPhase(Key key, boolean lastPhase)
                 ("Key must be an instance of PublicKey");
         }
         /* attempt to translate the key if it is not an ECKey */
-        this.publicKey = (ECPublicKeyImpl) ECKeyFactory.toECKey(key);
+        ECPublicKey ecKey = (ECPublicKey) ECKeyFactory.toECKey(key);
+        if (ecKey instanceof ECPublicKeyImpl) {
+            this.publicKey = (ECPublicKeyImpl) ecKey;
```

Comment:
The cast to `ECPublicKey` is unnecessary; also suggest using pattern matching language feature:
```
        ECKey ecKey = ECKeyFactory.toECKey(key);
        if (ecKey instanceof ECPublicKeyImpl keyImpl) {
            this.publicKey = keyImpl;
```